### PR TITLE
Moved dependencies to pyproject

### PR DESCRIPTION
**Pull request by sfc-gh-astus**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [ ] I've added new automated tests to verify correctness of my new code.
   * [x] I've described my changes in the section below.

### Changes description
Moved dependencies to pyproject to have all dependencies in one place, instead of 3.
To install dependencies run `pip install .`
To install dependencies with test dependencies run `pip install ".[tests]"`

### CODE REVIEW

**Review by sfc-gh-jsikorski — APPROVED** [review]

File: pyproject.toml [hunk 1]
```diff
@@ -33,6 +51,14 @@ classifiers = [
   "Topic :: Database"
 ]
 
+[project.optional-dependencies]
+tests = [
```
Comment (sfc-gh-turbaszek) [inline pyproject.toml:55]:
```suggestion
dev = [
```
Maybe more broad due to pre-commit?

File: pyproject.toml [hunk 2]
```diff
@@ -12,14 +12,32 @@ requires-python = ">=3.8"
...
+  "requests==2.28.1",
   "requirements-parser",
   "rich",
-  "snowflake-connector-python[secure-local-storage]",
+  "snowflake-connector-python==3.0.3",
   "toml",
-  "typer"
+  "typer",
+  "typing_extensions==4.5.0",
+  "urllib3==1.26.12",
```
[the remark this answers is not in the document]
Comment (sfc-gh-astus) [reply pyproject.toml:40]: list reduced

File: .github/workflows/build.yaml [hunk 3]
```diff
@@ -63,7 +63,7 @@ jobs:
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
-          pip install -r requirements.txt
+          pip install .
```
Comment (sfc-gh-turbaszek) [inline .github/workflows/build.yaml:66]: If we are working on deps should we add here `pip check` step? https://pip.pypa.io/en/stable/cli/pip_check/

**Review by sfc-gh-turbaszek — APPROVED** [review]
Small comment that can be addressed later on, otherwise looks good to me 👍

**Review by sfc-gh-turbaszek — APPROVED** [review]

**Pull request merged**